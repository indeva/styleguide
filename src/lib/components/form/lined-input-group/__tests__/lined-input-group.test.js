import React from 'react';
import renderer from 'react-test-renderer';
import LinedInputGroup from '..';

describe('LinedInputGroup', () => {
  it('renders the LinedInputGroup', () => {
    const component = renderer.create(
      <LinedInputGroup
        fieldRef={jest.fn()}
        id="id"
        name="name"
        label="Name 1"
        value="Test"
        onChange={jest.fn()}
        onBlur={jest.fn()}
      />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe('with custom class', () => {
    it('renders the LinedInputGroup', () => {
      const component = renderer.create(
        <LinedInputGroup
          id="id"
          name="name"
          label="Name 1"
          value="Test"
          onChange={jest.fn()}
          onBlur={jest.fn()}
          className="my class"
        />,
      );
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
