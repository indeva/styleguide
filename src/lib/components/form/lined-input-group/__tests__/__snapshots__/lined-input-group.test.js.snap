// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinedInputGroup renders the LinedInputGroup 1`] = `
<div
  className="form-group lined-input-group"
>
  <div
    className="lined-input"
  >
    <input
      className="lined-input__field"
      disabled={false}
      id="id"
      name="name"
      onBlur={[MockFunction]}
      onChange={[MockFunction]}
      required={true}
      type="text"
      value="Test"
    />
    <label
      className="lined-input__label"
      htmlFor="id"
    >
      Name 1
    </label>
    <div />
  </div>
  <span
    className="form-group__message"
  />
</div>
`;

exports[`LinedInputGroup with custom class renders the LinedInputGroup 1`] = `
<div
  className="form-group lined-input-group my class"
>
  <div
    className="lined-input"
  >
    <input
      className="lined-input__field"
      disabled={false}
      id="id"
      name="name"
      onBlur={[MockFunction]}
      onChange={[MockFunction]}
      required={true}
      type="text"
      value="Test"
    />
    <label
      className="lined-input__label"
      htmlFor="id"
    >
      Name 1
    </label>
    <div />
  </div>
  <span
    className="form-group__message"
  />
</div>
`;
